docs/syscall.md

```diff
@@ -27,15 +27,32 @@ These system calls are solely for the convenience of accessing the [SDL library]
 
 If a window does not already exist, one will be created with the specified `width` and `height`. The `screen` buffer will replace the content of the framebuffer, passing a different `width` or `height` compared to the size of the window is undefined behavior. This system call additionally polls events from the SDL library, and, if necessary, update the internal input specific event queue.
 
-### `poll_event(event)` - Poll a input specific event from SDL
+### `setup_queue` - Setup input system's dedicated event and submission queue
 
 **system call number**: `0xC0DE`
 
-**synopsis**: `int poll_event(void *event)`
+**synopsis**: `void *setup_queue(int capacity, unsigned int* event_count)`
 
-`event` will be filled with the polled event data, it should have a 32-bit `type` field, and associated with an appropriately sized value buffer. The internal event queue will be updated everytime `draw_frame` is called.
+Allocate a continuous memory chunk that has the requested capacity and two closely packed queues, the event queue and the submission queue. The base address of the event queue is located in the returned address, and the submission queue is immediately after the event queue's last element, which is the event queue's base address plus the size of each event element times the given capacity. The provided event counter variable is used to inform the user that an event is pushed to the event queue, its value will be increased whenever there is a new event, so initializing it before passing its address to this system call is critical.
```

Comment:
Rephrase as following:
> It is crucial to initialize the event counter variable before supplying its address to this system call because it serves as an interaction to the user that an event has been added to the event queue.